drivers/rpmsg/Kconfig

```diff
@@ -15,6 +15,16 @@ config RPMSG_CHAR
 	  in /dev. They make it possible for user-space programs to send and
 	  receive rpmsg packets.
 
+config RPMSG_CLK
+	tristate "RPMSG clock API redirection"
+	depends on COMMON_CLK
+	depends on RPMSG
+	help
+	  Say Y here to redirect clock API from the remote processor.
+	  With this driver, the remote processor could:
+	  1.Reuse the clock driver in the kernel side, or
+	  2.Form a hybrid(kernel plus RTOS) clock tree.
+
```

Comment:
I just don't get what this does. No idea...

Reply:
SOF defines a simple clock API here;
sof/src/include/sof/clock.h
Since all clocks normally form a tree structure, the leaf clock frequency decide by:
1.The parent clock frequency
2.The current clock hardware setting
But the parent clock instance normally put in to the kernel side, DSP/MCU could use this driver to get the parenet clock frequecny from kernel to avoid the hard code value.
Sometime we have to change the parent clock frequency, this driver can do so too. 
![clk](https://user-images.githubusercontent.com/18066964/47406304-28830780-d788-11e8-92d7-53f02e9a5e51.JPG)

We can even move MCU/DSP leaf clock definition to kernel side to reuse kernel clock infrastructure as much as possible.

Reply:
Humm, that's ambitious. the platforms I know have a lot of restrictions on how clock changes are done, and to synchronize the operations between what the RTOS requests and the kernel does is no small task.

Reply:
Yes, it's not a trival task. Our clk driver expert spend several months to achieve the  above picture.